File: 349-intersection-of-two-arrays/349-intersection-of-two-arrays.cpp
class Solution {
public:
    vector<int> intersection(vector<int>& nums1, vector<int>& nums2) {
        map<int, bool> mp;
        vector<int> ans;
        int n = nums1.size();
        
        for(int i = 0; i < n; i++) {
            mp[nums1[i]] = true;
        }
        
        n = nums2.size();
        
        for(int i = 0; i < n; i++) {
            if(mp[nums2[i]]) {
                ans.push_back(nums2[i]);
                mp[nums2[i]] = false;
            }
        }
        
        return ans;
    }
};

File: 349-intersection-of-two-arrays/349-intersection-of-two-arrays.js
/**
 * @param {number[]} nums1
 * @param {number[]} nums2
 * @return {number[]}
 */
var intersection = function(nums1, nums2) {
    let set = new Set();
    nums1.map(val => set.add(val));
    let ans = [];
    
    nums2.map(val => {
        if(set.has(val)) {
            set.delete(val);
            ans.push(val);
        }
    })
    
    return ans;
};